### PR TITLE
fix: konflux: always rebase before tests run

### PR DESCRIPTION
**Pull request by psturc**
https://issues.redhat.com/browse/KFLUXBUGS-1185

* related changes are in https://github.com/redhat-appstudio/e2e-tests/pull/1172

### CODE REVIEW

File: ci-operator/step-registry/redhat-appstudio/load-test/redhat-appstudio-load-test-commands.sh [hunk 1]
```diff
@@ -89,13 +89,14 @@ echo "https://${GITHUB_USER}:${GITHUB_TOKEN}@github.com" >"${GIT_CREDS_PATH}"
...
-git clone --branch main "https://${GITHUB_TOKEN}@github.com/redhat-appstudio/e2e-tests.git" .
+git clone --origin upstream --branch main "https://${GITHUB_TOKEN}@github.com/redhat-appstudio/e2e-tests.git" .
 
 set -x
 if [ "$JOB_TYPE" == "presubmit" ] && [[ "$JOB_NAME" != rehearse-* ]]; then
     # if this is executed as PR check of github.com/redhat-appstudio/e2e-tests.git repo, switch to PR branch.
-    git fetch origin "pull/${PULL_NUMBER}/head"
+    git fetch upstream "pull/${PULL_NUMBER}/head"
     git checkout -b "pr-${PULL_NUMBER}" FETCH_HEAD
+    git pull --rebase upstream main
```
Comment (psturc) [inline ci-operator/step-registry/redhat-appstudio/load-test/redhat-appstudio-load-test-commands.sh:99]: Hi @jhutar could you PTAL at changes to load-test scripts?

**Comment by psturc** [comment]
/pj-rehearse pull-ci-redhat-appstudio-e2e-tests-main-redhat-appstudio-e2e pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests pull-ci-redhat-appstudio-infra-deployments-main-appstudio-upgrade-tests

**Comment by psturc** [comment]
[infra-deployments](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/52146/rehearse-52146-pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests/1791387992087072768) and [e2e-tests](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/52146/rehearse-52146-pull-ci-redhat-appstudio-e2e-tests-main-redhat-appstudio-e2e/1791387991978020864) jobs succeeded

retesting upgrade job now:
/pj-rehearse pull-ci-redhat-appstudio-infra-deployments-main-appstudio-upgrade-tests

**Comment by psturc** [comment]
/pj-rehearse pull-ci-redhat-appstudio-infra-deployments-main-appstudio-upgrade-tests

**Comment by psturc** [comment]
/pj-rehearse pull-ci-redhat-appstudio-infra-deployments-main-appstudio-upgrade-tests

**Review by jhutar — APPROVED** [review]
/lgtm

File: ci-operator/step-registry/redhat-appstudio/load-test/redhat-appstudio-load-test-commands.sh (hunk 1 above)
Comment (psturc) [reply ci-operator/step-registry/redhat-appstudio/load-test/redhat-appstudio-load-test-commands.sh:101]: @jhutar thanks for review! I realized I need to add the `origin` remote again due to keeping the backward compatibility before [this PR](https://github.com/redhat-appstudio/e2e-tests/pull/1172/files#diff-293f6cf70e46616af76ba8cde40b57ce82d2cbbf7b632c10da79fd36294a933cL38) gets merged - would you mind giving it another look please? Thanks!

**Comment by psturc** [comment]
/pj-rehearse pull-ci-redhat-appstudio-e2e-tests-main-redhat-appstudio-e2e pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests pull-ci-redhat-appstudio-infra-deployments-main-appstudio-upgrade-tests

File: ci-operator/step-registry/redhat-appstudio/load-test/redhat-appstudio-load-test-commands.sh (hunk 1 above)
Comment (jhutar) [reply ci-operator/step-registry/redhat-appstudio/load-test/redhat-appstudio-load-test-commands.sh:101]: Oh, nice!

**Review by jhutar — APPROVED** [review]
/lgtm

**Pull request merged**